docs/source/inference.rst

```diff
@@ -0,0 +1,531 @@
+    tokenizer = AutoTokenizer.from_pretrained(model_id)
+
+    # Push to hub
+    USER_ID = "YOUR_USER_ID"
+    MODEL_NAME = model_id.split("/")[-1]
+    save_to = f"{USER_ID}/{MODEL_NAME}-float8dq"
+    quantized_model.push_to_hub(save_to, safe_serialization=False)
+    tokenizer.push_to_hub(save_to)
+
+    # Manual Testing
+    prompt = "Hey, are you conscious? Can you talk to me?"
+    messages = [
+        {
+            "role": "system",
+            "content": "",
+        },
+        {"role": "user", "content": prompt},
+    ]
+    templated_prompt = tokenizer.apply_chat_template(
+        messages,
+        tokenize=False,
+        add_generation_prompt=True,
+    )
+    print("Prompt:", prompt)
+    print("Templated prompt:", templated_prompt)
+    inputs = tokenizer(
+        templated_prompt,
+        return_tensors="pt",
+    ).to("cuda")
+    generated_ids = quantized_model.generate(**inputs, max_new_tokens=128)
+    output_text = tokenizer.batch_decode(
+        generated_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False
+    )
+    print("Response:", output_text[0][len(prompt):])
+
+
+[Optional] Float8 Dynamic Quantization + Semi-structured (2:4) sparsity
+--------------------------------------------------------------------------
+
+Torchao's sparsity support can be combined with quantization for additional performance gains, using optimized kernels for 2:4 structured sparsity.
+
+.. code-block:: python
+
+    from torchao.quantization import Float8DynamicActivationFloat8SemiSparseWeightConfig
+
+    # Combine sparsity with int4 quantization - both optimized by torchao
+    quant_config = Float8DynamicActivationFloat8SemiSparseWeightConfig()
+    quantization_config = TorchAoConfig(quant_type=quant_config)
+
+    # Load a pre-sparsified checkpoint
+    model = AutoModelForCausalLM.from_pretrained(
+        "RedHatAI/Sparse-Llama-3.1-8B-2of4",  # 2:4 sparse model
+        torch_dtype=torch.float16,
+        device_map="cuda",
+        quantization_config=quantization_config
+    )
+
+    tokenizer = AutoTokenizer.from_pretrained("meta-llama/Llama-3.1-8B-Instruct")
+
+    # Use static KV cache for best performance with torchao optimizations
+    messages = [{"role": "user", "content": "What are the benefits of sparse neural networks?"}]
+    inputs = tokenizer.apply_chat_template(messages, return_tensors="pt", add_generation_prompt=True).to("cuda")
+
+    outputs = model.generate(
+        inputs,
+        max_new_tokens=150,
+        cache_implementation="static",  # Optimized for torchao
+        do_sample=False
+    )
+
+    response = tokenizer.decode(outputs[0][inputs.shape[1]:], skip_special_tokens=True)
+    print(response)
+
+.. note::
+    For more information on supported quantization and sparsity configurations, see `HF-Torchao Docs <https://huggingface.co/docs/transformers/main/en/quantization/torchao>`_.
+
+Inference with vLLM
```

Comment:
for this section, can you replace with https://huggingface.co/pytorch/Qwen3-8B-int4wo-hqq#inference-with-vllm

it might be easier to do command line compared to code